README.md

```diff
@@ -1,21 +1,67 @@
 # Common files for os-autoinst/os-autoinst and os-autoinst/openQA
 
-This repository is to be used as a
-[git-subrepo](https://github.com/ingydotnet/git-subrepo).
+This repository is to be used as a [git-subrepo]
+(https://github.com/ingydotnet/git-subrepo). See the instructions below in the
+[Git Subrepo Usage](#git-subrepo-usage) section
 
+All dependencies in this project are sourced from what's available in OpenSUSE
+Tumbleweed RPM repositories.
+
+For developers not using RPM the tumbleweed repositories, a `cpanfile` is
+provided and maintained in a best-effort basis.
+
+## Tooling offered
+
+This repo offers:
+
+* Linter configuration for perl projects: `.perltidyrc`, `.perlcriticrc`,
+  `.proverc` & `.tidyallrc`.
+* Perlcritic policies that `os-autoinst/os-autoinst` and `os-autoinst/openQA`
+  share.
+* Useful tools for linting & testing:
+    * A `perlcritic` wrapper that will automatically add Perlcritic rules
+      defined under `lib/perlcritic` and
+      `external/os-autoinst-common/lib/perlcritic`.
+    * A `tidyall` wrapper that will validate the Perltidy version against the
+      `cpanfile` definition.  
+      Because `Perl::Tidy` defaults may vary between versions this tool ensures
+      the version of perltidy matches the required version specified in the
+      `cpanfile` for non-`cpanm` based installs.
+
+* Example Github Actions for linting and unit testing for Perl.
+
+All files can be either copied or symlinked for any downstream repository
+consuming these tools.
+
+## Running tools and tests from scratch
+
+```bash
+# Run a container with the project mounted in it.
+podman run -it -v "$PWD:/host/project" --workdir /host/project opensuse/leap:latest bash
+
+# Inside the container
+zypper in -y perl-App-cpanminus make gcc
+cpanm --installdeps . --with-develop
+
+prove .
+./tools/tidyall --check-only --all --quiet
+./tools/perlcritic --quiet .
```

Comment:
I strongly prefer a simple Makefile that just offers to call all that with a simple `make test`. If you call the Makefile in GHA or use the low-level commands I consider less important although my personal preference is that CI sticks as much as possible to what we expect to work for users

Reply:
You'll probably want to call the exact same tools in all projects? I can see that you don't want to try and unify everything.

How about you just add a new target that does whatever perl-lint-checks is supposed to do. Don't even try to change anything else.

Reply:
Yes, I was also only thinking of a Makefile that does conduct the tests in that repo. Makefile might act as an example for other repos but does not need to.